pkg/controller/installation/windows_controller.go

```diff
@@ -0,0 +1,597 @@
+		reqLogger.V(2).Info("loaded final computed config", "config", instance)
+
+		// Validate the configuration.
+		if err := validateCustomResource(instance); err != nil {
+			r.status.SetDegraded(operatorv1.InvalidConfigurationError, "Invalid computed config", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// A status is needed at this point for operator scorecard tests.
+	// status.variant is written later but for some tests the reconciliation
+	// does not get to that point.
+	if reflect.DeepEqual(instanceStatus, operatorv1.InstallationStatus{}) {
+		instance.Status = operatorv1.InstallationStatus{}
+		if err := r.client.Status().Update(ctx, instance); err != nil {
+			r.status.SetDegraded(operatorv1.ResourceUpdateError, "Failed to write default status", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+	}
+
+	// The operator supports running in a "Calico only" mode so that it doesn't need to run TSEE specific controllers.
+	// If we are switching from this mode to one that enables TSEE, we need to restart the operator to enable the other controllers.
+	if !r.enterpriseCRDsExist && instance.Spec.Variant == operatorv1.TigeraSecureEnterprise {
+		// Perform an API discovery to determine if the necessary APIs exist. If they do, we can reboot into TSEE mode.
+		// if they do not, we need to notify the user that the requested configuration is invalid.
+		b, err := utils.RequiresTigeraSecure(r.config)
+		if b {
+			logw.Info("Rebooting to enable TigeraSecure controllers")
+			os.Exit(0)
+		} else if err != nil {
+			r.status.SetDegraded(operatorv1.InternalServerError, "Error discovering Tigera Secure availability", err, reqLogger)
+		} else {
+			r.status.SetDegraded(operatorv1.InternalServerError, "Cannot deploy Tigera Secure", fmt.Errorf("Missing Tigera Secure custom resource definitions"), reqLogger)
+		}
+
+		// Queue a retry. We don't want to watch the APIServer API since it might not exist and would cause
+		// this controller to fail.
+		reqLogger.Info("Scheduling a retry in 30 seconds")
+		return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+	}
+
+	// The operator supports running without the AmazonCloudIntegration when it's CRD is not installed.
+	// If, when this controller was started, the CRD didn't exist, but it does now, then reboot.
+	if !r.amazonCRDExists {
+		amazonCRDRequired, err := utils.RequiresAmazonController(r.config)
+		if err != nil {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Error discovering AmazonCloudIntegration CRD", err, reqLogger)
+			reqLogger.Info("Scheduling a retry in 30 seconds")
+			return reconcile.Result{RequeueAfter: 30 * time.Second}, nil
+		}
+		if amazonCRDRequired {
+			logw.Info("Rebooting to enable AWS controllers")
+			os.Exit(0)
+		}
+	}
+
+	certificateManager, err := certificatemanager.Create(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace(), certificatemanager.AllowCACreation())
```

Comment:
We shouldn't have `AllowCACreation` - the cluster CA is created by the main secrets controller and all other controllers should just wait for it.